index.js

```diff
@@ -26,19 +29,30 @@ module.exports = function (src, dest, opts) {
 
 	opts = objectAssign({overwrite: true}, opts);
 
-	return new Promise(function startRead(resolve, reject) {
-		var read = fs.createReadStream(src);
-
-		read.on('error', function (err) {
-			reject(new CpFileError('cannot read from `' + src + '`: ' + err.message, err));
-		});
+	var progress = {
+		stat: null,
+		emitter: new EventEmitter()
+	};
```

Comment:
> , why check?

I do not know if I have understood you correctly.

I think we should always `fs.lstat` to determine the size. The alternative will be to postpone the call until a listener is added, which I think is too cumbersome.

Checking `emitter.listenerCount('progress') > 0` before emitting is double workload, `emitter.emit` does a similar check.

Checking `size !== null` before emitting was based on a misconception (mentioned below), which have worked well in case of `opts.onProgress` :unamused:

Hence, I don't think we should check before emitting progress at all.